Index: tests/libappimage/CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- tests/libappimage/CMakeLists.txt	(revision 0dd6cf369a4bc066ea2dff6bc5795a948a4b9364)
+++ tests/libappimage/CMakeLists.txt	(date 1559878340000)
@@ -21,6 +21,6 @@
 )
 
 target_include_directories(test_libappimage++ PRIVATE "${PROJECT_SOURCE_DIR}/src/libappimage")
-target_link_libraries(test_libappimage++ libappimage Boost::filesystem libarchive libsquashfuse XdgUtils::DesktopEntry XdgUtils::BaseDir gtest gtest_main)
+target_link_libraries(test_libappimage++ libappimage Boost::filesystem libarchive libsquashfuse XdgUtils::DesktopEntry XdgUtils::BaseDir GTest::GTest GTest::Main)
 
 add_test(test_libappimage++ test_libappimage++)
Index: tests/libappimage/legacy/CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- tests/libappimage/legacy/CMakeLists.txt	(revision 0dd6cf369a4bc066ea2dff6bc5795a948a4b9364)
+++ tests/libappimage/legacy/CMakeLists.txt	(date 1559878340000)
@@ -1,9 +1,9 @@
 add_library(fixtures INTERFACE)
 target_sources(fixtures INTERFACE ${CMAKE_CURRENT_SOURCE_DIR}/fixtures.h)
-set_property(TARGET fixtures PROPERTY INTERFACE_LINK_LIBRARIES xdg-basedir gtest)
+set_property(TARGET fixtures PROPERTY INTERFACE_LINK_LIBRARIES xdg-basedir GTest::GTest)
 
 add_executable(test_libappimage test_libappimage.cpp)
-target_link_libraries(test_libappimage fixtures libappimage libsquashfuse libglib libgobject gtest gtest_main)
+target_link_libraries(test_libappimage fixtures libappimage libsquashfuse libglib libgobject GTest::GTest GTest::Main)
 add_test(test_libappimage test_libappimage)
 
 add_executable(test_shared test_shared.cpp)
Index: tests/CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- tests/CMakeLists.txt	(revision 0dd6cf369a4bc066ea2dff6bc5795a948a4b9364)
+++ tests/CMakeLists.txt	(date 1559878340000)
@@ -3,6 +3,9 @@
 if(BUILD_TESTING)
     cmake_minimum_required(VERSION 3.5)
 
+    if(USE_CONAN)
+        find_package(GTest REQUIRED)
+    endif()
     # global definitions
     add_definitions(
         -DTEST_DATA_DIR="${CMAKE_CURRENT_SOURCE_DIR}/data/"
Index: cmake/dependencies.cmake
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- cmake/dependencies.cmake	(revision 0dd6cf369a4bc066ea2dff6bc5795a948a4b9364)
+++ cmake/dependencies.cmake	(date 1559878340000)
@@ -6,6 +6,11 @@
 # imported dependencies
 include(${CMAKE_CURRENT_LIST_DIR}/imported_dependencies.cmake)
 
+if(USE_CONAN)
+    include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+    conan_basic_setup(NO_OUTPUT_DIRS)
+endif()
+
 if(USE_CCACHE)
     message(STATUS "Using CCache to build AppImageKit dependencies")
     # TODO: find way to use find_program with all possible paths
Index: lib/CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- lib/CMakeLists.txt	(revision 0dd6cf369a4bc066ea2dff6bc5795a948a4b9364)
+++ lib/CMakeLists.txt	(date 1559878340000)
@@ -3,11 +3,13 @@
 include(CTest)
 
 if(BUILD_TESTING)
-    if(NOT TARGET gtest)
-        if(IS_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/gtest)
-            add_subdirectory(gtest EXCLUDE_FROM_ALL)
-        else()
-            message(FATAL_ERROR "gtest submodule not found; please call git submodule update --init or disable the unit tests using -DBUILD_TESTING=OFF")
+    if(NOT USE_CONAN)
+        if(NOT TARGET gtest)
+            if(IS_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/gtest)
+                add_subdirectory(gtest EXCLUDE_FROM_ALL)
+            else()
+                message(FATAL_ERROR "gtest submodule not found; please call git submodule update --init or disable the unit tests using -DBUILD_TESTING=OFF")
+            endif()
         endif()
     endif()
 endif()
Index: tests/libappimage/desktop_integration/CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- tests/libappimage/desktop_integration/CMakeLists.txt	(revision 0dd6cf369a4bc066ea2dff6bc5795a948a4b9364)
+++ tests/libappimage/desktop_integration/CMakeLists.txt	(date 1559878340000)
@@ -39,10 +39,13 @@
     # statically once glib is loaded.
     PRIVATE libglib
     PUBLIC dl
-    PRIVATE gtest
-    PRIVATE gtest_main
+    PRIVATE GTest::GTest
+    PRIVATE GTest::Main
     PRIVATE librsvg
     PRIVATE libcairo
 )
 
+get_target_property(GTEST_PROP GTest::GTest INTERFACE_LINK_LIBRARIES)
+MESSAGE("GTEST_PROP  ${GTEST_PROP}")
+
 add_test(TestDesktopIntegration TestDesktopIntegration)
Index: CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- CMakeLists.txt	(revision 0dd6cf369a4bc066ea2dff6bc5795a948a4b9364)
+++ CMakeLists.txt	(date 1559878340000)
@@ -32,6 +32,7 @@
 option(LIBAPPIMAGE_THUMBNAILER_ENABLED "Enable thumbnailer functions" On)
 option(LIBAPPIMAGE_STANDALONE "Statically embbed dependencies" Off)
 option(ENABLE_COVERAGE "Enable tests code coverate target" Off)
+option(USE_CONAN "Use conan to resolve dependencies" Off)
 
 
 if(ENABLE_COVERAGE)
